src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -274,8 +274,18 @@ def __init__(
         self._uuid = str(uuid.uuid4())
         # Metadata/Attributes for the plan
         self._attributes: Optional[List[Attribute]] = None
+        self._quoted_identifiers: Optional[List[str]] = None
+        # If _attributes is not None, then _quoted_identifiers will be None.
+        # If _quoted_identifiers is not None, then _attributes will be None.
         if session.reduce_describe_query_enabled and self.source_plan is not None:
-            self._attributes = infer_metadata(self.source_plan)
+            self._attributes, self._quoted_identifiers = infer_metadata(
+                self.source_plan,
+                self.session._analyzer,
+                self.df_aliased_col_name_to_real_col_name,
+            )
+        assert not (
```

Comment:
add a comment here about why we have such check here

Reply:
done